lib/spack/spack/solver/asp.py

```diff
@@ -2840,6 +2797,87 @@ def reject_requirement_constraint(
         return False
 
 
+class CompilerParser:
```

Comment:
Parser? :think

Reply:
I'd leave this as is, at least here. There is also a `RequirementParser` - open to change both names in a following Pr.